libraries/roomselect/impl/src/main/kotlin/io/element/android/libraries/roomselect/impl/RoomSelectSearchDataSource.kt

```diff
@@ -38,11 +38,20 @@ class RoomSelectSearchDataSource @Inject constructor(
         source = RoomList.Source.All,
     )
 
-    val roomSummaries: Flow<PersistentList<RoomSummary>> = roomList.filteredSummaries
+    val roomInfoList: Flow<PersistentList<SelectRoomInfo>> = roomList.filteredSummaries
         .map { roomSummaries ->
             roomSummaries
-                .filter { it.currentUserMembership == CurrentUserMembership.JOINED }
+                .filter { it.info.currentUserMembership == CurrentUserMembership.JOINED }
                 .distinctBy { it.roomId } // This should be removed once we're sure no duplicate Rooms can be received
+                .map { roomSummary ->
+                    SelectRoomInfo(
+                        roomId = roomSummary.roomId,
+                        name = roomSummary.info.name,
+                        avatarUrl = roomSummary.info.avatarUrl,
+                        heroes = roomSummary.info.heroes,
+                        canonicalAlias = roomSummary.info.canonicalAlias,
+                    )
+                }
```

Comment:
Move function `toSelectRoomInfo()` from the test class to this file and use it here?